go/libraries/doltcore/sqle/schema_table.go

```diff
@@ -69,6 +70,20 @@ var schemaTableSchema = schema.MustSchemaFromCols(schemasTableCols)
 // getOrCreateDoltSchemasTable returns the `dolt_schemas` table in `db`, creating it if it does not already exist.
 // Also migrates data to the correct format if necessary.
 func getOrCreateDoltSchemasTable(ctx *sql.Context, db Database) (retTbl *WritableDoltTable, retErr error) {
+	root, err := db.GetRoot(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	schemaName := doltdb.DefaultSchemaName
+	if resolve.UseSearchPath && db.schemaName == "" {
+		schemaName, err = resolve.FirstExistingSchemaOnSearchPath(ctx, root)
+		if err != nil {
+			return nil, err
+		}
+		db.schemaName = schemaName
+	}
```

Comment:
So if `db.schemeName` has a value (such as `example_schema`), then we're never setting that schema? That's what it looks like from this section at least. We start with `doltdb.DefaultSchemaName`, and then resolve the schema name only if `db.schemaName == ""`. So if `db.schemaName != ""`, then we never assign and use it.